### ISSUE TITLE
Unable to remote SSH Backup with Ubuntu 16.04

### ISSUE THREAD

**Issue by wiserweb** [webmin/webmin#439]
Issue:

It seems there might be something broken with backups on Ubuntu 16.04. 

Background:

We have a remote backup server that is accessible via ssh on a custom public port. The machine itself is Ubuntu 16.04.

The new Virtualmin box that we are moving to is also running Ubuntu 16.04.

I copied the exact configuration from the Debian Wheezy box to the Ubuntu 16.04 box. However the backups to not want to complete.

I am able to connect to the backup box using ssh with a normal user. I have used ssh-copy-id to copy the non-root user public key. I have also generated ssh key for the root user and have used ssh-copy-id to copy the public key to the backup server (just in case).  I am able to connect to backup server using either user via plain ssh command as expected.

i.e.
```
  "ssh -p 'port' 'user@IPADDRESS'"
```

On the Ubuntu 16.05 host using the Virtualmin GUI the following error is displayed when remote SSH backup location is specified.

```
Starting backup of 1 domains to /home/owner/domain/%Y-%m-%d on SSH server IPADDRESS ..
Failed to connect to SSH server : owner@IPADDRESS's password: Permission denied, please try again. owner@IPADDRESS's password: Permission denied, please try again. owner@IPADDRESS's password: Permission denied (publickey,password). lost connection 
Backup failed! See the progress output above for the reason why.
```

Tried the following virtualmin CLI command in attempt to get better debug output.

Using root user from command line

`
virtualmin backup-domain --domain domain.com --all-features --dest ssh://USER@IPADDRESS:PORT:/home/owner/domain/domain.tgz`
`

Result
```
Failed to open /usr/share/webmin/virtual-server/ssh://owner@IPADDRESS:PORT:/home/owner/DOMAIN/support.tgz for writing : No such file or directory at ../web-lib-funcs.pl line 1427.
```

Using non root user

`
sudo virtualmin backup-domain --domain domain.com --all-features --dest ssh://USER@IPADDRESS:PORT:/home/owner/domain/domain.tgz`
`

Result (same error)

```
Failed to open /usr/share/webmin/virtual-server/ssh://owner@IPADDRESS:PORT:/home/owner/DOMAIN/support.tgz for writing : No such file or directory at ../web-lib-funcs.pl line 1427.
```

Using the same command on Debian Wheezy results in successful backup.

```
.. completed in 23 seconds
1 servers backed up successfully, 0 had errors. 
Backup is complete. Final size was 15.05 MB.

```

Verifying the logs on the backup server using `tail /var/log/auth.log` shows that no connection attempts from Virtualmin Ubuntu 16.04 host are recorded. 

So this leads me to think if SSH can connect but Virtualmin cannot, it might be related to Virtualmin Ubuntu 16.04 as the Debian Wheezy performs the expected behaviour without any errors.

Thanks.

**Comment by wiserweb** [comment]
[enhancing this ticker with STR]
Steps to Reproduce:

[Host System]
1. Install fresh Ubuntu 16.04 with Virtualmin
2. Ensure root user can operate as root with password, (sudo /usr/share/webmin/changepass.pl /etc/webmin root [yourpassword-here])
3. ssh-keygen for normal user and root user
4. ssh-copy-id normal user and root user to Remote Backup System
5. Verify ssh passwordless logins work
6. Setup Remote Backup task to Remote Backup System with SSH using non-standard public port, specify remote location as /home/user/backups (see step 3 below)
7. Manually initiate Remote Backup task

[Remote Backup System]
1. Install fresh Ubuntu 16.04
2. Ensure normal user has expected public keys in .ssh/authorized_keys
3. Create backup folder location in user home, i.e. /home/user/backups

**Comment by jcameron** [comment]
If there is no password needed (because you have an SSH key setup), you need to use a URL like `ssh://USER:@IPADDRESS:PORT:/home/owner/domain/domain.tgz`

**Comment by wiserweb** [comment]
Ok... confirmed. Adding the colon after the USER parameter allows the backup-domain command to complete successfully.

sudo virtualmin backup-domain --domain domain.com --all-features --dest ssh://USER:@IPADDRESS:PORT:/home/owner/support.tgz

Now onto the GUI. The strangest thing Jamie... I haven't touched the server for a few days (since I created this issue). I tried the backup from the Webmin GUI moments ago and it now it works as expected.

It's still not clear to me what sort of issue I was experiencing. At this point I cannot replicate the error I was receiving. I think this was likely caused by a network related issue rather than webmin. 

I will close this issue.

Thanks for your help.
